### PR TITLE
[10.0] ADD website_sale_charge_payment_fee

### PR DESCRIPTION
**Pull request by eLBati**
modules
```website_sale_charge_payment_fee```
```website_sale_charge_payment_fee_delivery```

### CODE REVIEW

File: website_sale_charge_payment_fee/__manifest__.py
Comment (MeetKD) [inline website_sale_charge_payment_fee/__manifest__.py]: Can you pls remove empty key?

**Review by MeetKD — CHANGES_REQUESTED** [review]

**Comment by eLBati** [comment]
@MeetKD done

File: website_sale_charge_payment_fee/controllers/website_sale.py
Comment (SimoRubi) [inline website_sale_charge_payment_fee/controllers/website_sale.py:11]: Shouldn't the route be `@http.route(['/shop/payment'], type='http', auth="public", website=True)` ?

**Review by hurrinico — APPROVED** [review]
👍 LGTM

**Review by SimoRubi — COMMENTED** [review]
Looks good, waiting for runbot

**Review by Sashimi22 — APPROVED** [review]
Tested functionally 👍

File: website_sale_charge_payment_fee/controllers/website_sale.py
Comment (yajo) [reply website_sale_charge_payment_fee/controllers/website_sale.py:11]: If the decorator was declared in the parent method, you don't need to repeat things.

**Comment by SimoRubi** [comment]
👍 Thanks, didn't know that

File: website_sale_charge_payment_fee/controllers/website_sale.py
Comment (SimoRubi) [reply website_sale_charge_payment_fee/controllers/website_sale.py:11]: Thanks, didn't know that, I usually keep repeating decorators in every overriding method.

Comment (yajo) [reply website_sale_charge_payment_fee/controllers/website_sale.py:11]: Yes, you need the decorator; just not the arguments.

**Review by MeetKD — APPROVED** [review]

File: website_sale_charge_payment_fee/models/sale.py
Comment (simahawk) [inline website_sale_charge_payment_fee/models/sale.py:28]: what if the line is already there? Wouldn't be better to search for it via product?

File: website_sale_charge_payment_fee/static/src/js/website_sale_fee.js
Comment (simahawk) [inline website_sale_charge_payment_fee/static/src/js/website_sale_fee.js:1]: I'd switch this to snippet animation

File: website_sale_charge_payment_fee/static/src/js/website_sale_fee.js
Comment (simahawk) [inline website_sale_charge_payment_fee/static/src/js/website_sale_fee.js]: what if you have extra args in qstring to keep?

File: website_sale_charge_payment_fee/static/src/js/website_sale_fee.js
Comment (simahawk) [inline website_sale_charge_payment_fee/static/src/js/website_sale_fee.js]: ?

File: website_sale_charge_payment_fee/views/website_sale_view.xml
Comment (simahawk) [inline website_sale_charge_payment_fee/views/website_sale_view.xml]: /views/ - > /templates/

File: website_sale_charge_payment_fee/views/website_sale_view.xml
Comment (simahawk) [inline website_sale_charge_payment_fee/views/website_sale_view.xml]: why closing the span here? you can nest them

File: website_sale_charge_payment_fee_delivery/models/delivery.py
Comment (simahawk) [inline website_sale_charge_payment_fee_delivery/models/delivery.py]: would be better to wrap the whole block inside `with self.env.do_in_draft()`

**Review by simahawk — CHANGES_REQUESTED** [review]

**Review by SimoRubi — APPROVED** [review]
Tested in runbot

**Comment by pedrobaeza** [comment]
Travis is red

File: website_sale_charge_payment_fee/models/sale.py
Comment (eLBati) [reply website_sale_charge_payment_fee/models/sale.py:28]: @simahawk previous ```payment_fee_line``` would be deleted by line 13

File: website_sale_charge_payment_fee/static/src/js/website_sale_fee.js
Comment (eLBati) [reply website_sale_charge_payment_fee/static/src/js/website_sale_fee.js:1]: What is it? Do you have an example? Consider I followed the same approach of ```website_sale_delivery```

File: website_sale_charge_payment_fee/static/src/js/website_sale_fee.js
Comment (eLBati) [reply website_sale_charge_payment_fee/static/src/js/website_sale_fee.js]: https://github.com/odoo/odoo/blob/2198220/addons/website_sale/static/src/js/website_sale_payment.js#L40

File: website_sale_charge_payment_fee/static/src/js/website_sale_fee.js
Comment (eLBati) [reply website_sale_charge_payment_fee/static/src/js/website_sale_fee.js]: I assume I will not have. Same behaviour of ```website_sale_delivery```

File: website_sale_charge_payment_fee/views/website_sale_view.xml
Comment (eLBati) [reply website_sale_charge_payment_fee/views/website_sale_view.xml]: done

File: website_sale_charge_payment_fee/views/website_sale_view.xml
Comment (eLBati) [reply website_sale_charge_payment_fee/views/website_sale_view.xml]: done

File: website_sale_charge_payment_fee_delivery/models/delivery.py
Comment (eLBati) [reply website_sale_charge_payment_fee_delivery/models/delivery.py]: done

**Comment by eLBati** [comment]
Travis :green_apple:

File: website_sale_charge_payment_fee/models/sale.py
Comment (simahawk) [reply website_sale_charge_payment_fee/models/sale.py:28]: missed it :stuck_out_tongue_closed_eyes:

File: website_sale_charge_payment_fee/static/src/js/website_sale_fee.js
Comment (simahawk) [reply website_sale_charge_payment_fee/static/src/js/website_sale_fee.js:1]: quick example https://github.com/OCA/website-cms/blob/10.0/cms_form/static/src/js/lock_copy_paste.js
but if you search JS files for 'web_editor.snippets.animation' you'll find tons of examples.
Not mandatory at all, especially since is a migration :wink:

File: website_sale_charge_payment_fee/static/src/js/website_sale_fee.js
Comment (simahawk) [reply website_sale_charge_payment_fee/static/src/js/website_sale_fee.js]: ok, but I don't get the comment and the commented line: is a TODO then? If so, add 'TODO', if not, remove these lines :wink:

File: website_sale_charge_payment_fee_delivery/models/delivery.py
Comment (simahawk) [reply website_sale_charge_payment_fee_delivery/models/delivery.py]: hmm I don't think is going to do what you expect the way you did it.
As soon as the ctx manager is exited you won't have `is_delivery` set and your call to `get_price_available` is going to be the same.
You should wrap the call inside it  and you don't need to set it back again ;)

**Review by simahawk — CHANGES_REQUESTED** [review]

File: website_sale_charge_payment_fee_delivery/models/delivery.py
Comment (eLBati) [reply website_sale_charge_payment_fee_delivery/models/delivery.py]: @simahawk thanks, done

File: website_sale_charge_payment_fee/static/src/js/website_sale_fee.js
Comment (eLBati) [reply website_sale_charge_payment_fee/static/src/js/website_sale_fee.js]: @simahawk improved comment: it is not a TODO, it is the reason for the presence of ```if (payment_id != selected_acquirer_id)```. Let me know if it is a bit more clear now

File: website_sale_charge_payment_fee/static/src/js/website_sale_fee.js
Comment (simahawk) [inline website_sale_charge_payment_fee/static/src/js/website_sale_fee.js]: @eLBati got it... "check" sounds more like "we have to check something" not that you have to tick/activate/enable a flag :wink: 
sounds better now, thanks!

**Review by simahawk — APPROVED** [review]

**Comment by eLBati** [comment]
Added ```website_sale_charge_payment_fee_quote``` to make ```website_sale_charge_payment_fee``` work with ```website_quote```, but travis, about this like https://github.com/eLBati/e-commerce/blob/d3d2c3ebebe823e8da865bfabd1149d87c41ef85/website_sale_charge_payment_fee/static/src/js/website_sale_fee.js#L4 , says ```u'TypeError: undefined is not a function (evaluating \'window.location.pathname.startsWith("/shop/payment")\')```
Does anybody know why?

**Comment by simahawk** [comment]
@eLBati not supported maybe? AFAIK it comes w/ ES6
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith
Was it working before or is a new thing?

**Comment by eLBati** [comment]
@simahawk right. Replaced with ```_.str.startsWith``` and it is green now

**Comment by eLBati** [comment]
@OCA/e-commerce-maintainers mergeable?

**Pull request merged**
